eng/packages/http-client-csharp/generator/TestProjects/Local/Mgmt-TypeSpec/src/Generated/FooResource.cs

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using MgmtTypeSpec.Models;
+
+namespace MgmtTypeSpec
+{
+    /// <summary></summary>
+    public partial class FooResource : ArmResource
+    {
+        private FooData _data;
+        private ClientDiagnostics _fooClientDiagnostics;
+        private Foos _fooRestClient;
+        /// <summary> Gets the resource type for the operations. </summary>
+        public static readonly ResourceType ResourceType = "MgmtTypeSpec/foos";
+
+        /// <summary> Initializes a new instance of FooResource for mocking. </summary>
+        protected FooResource()
+        {
+        }
+
+        internal FooResource(ArmClient client, FooData data) : this(client, data.Id)
+        {
+            HasData = true;
+            _data = data;
+        }
+
+        internal FooResource(ArmClient client, ResourceIdentifier id) : base(client, id)
+        {
+            _fooClientDiagnostics = new ClientDiagnostics("MgmtTypeSpec", ResourceType.Namespace, Diagnostics);
+            TryGetApiVersion(ResourceType, out string fooApiVersion);
+            _fooRestClient = new Foos(Pipeline, Endpoint, fooApiVersion);
+#if DEBUG
+            global::MgmtTypeSpec.FooResource.ValidateResourceId(id);
```

Comment:
Is this not getting reduced because its in an if debug?

Reply:
This is following the behavior of autorest.csharp, I don't have the knowledge of the actual intention here. https://github.com/Azure/autorest.csharp/blob/1338f40af8912aef1f8ee7a5c5167c311b66bcdb/test/TestProjects/MgmtTypeSpec/src/Generated/FooResource.cs#L68-L70
But yes, this only applies for debug.

Reply:
raise issue in MGC to fix this
we can replace it with Debug.Assert

Reply:
`Debug.Assert` can't be used here, it needs a bool condition input, but this method is void. 
Use `[Conditional("DEBUG")]` in the method signature instead.

Reply:
https://github.com/microsoft/typespec/issues/6160 to track the issue in MGC